readthedocs/core/unresolver.py

```diff
@@ -1,99 +1,284 @@
+        unresolve the path into a valid version of the project.
+
+        :param parent_project: The project that owns the path.
+        :param path: The path to unresolve.
+        :param check_subprojects: If we should check for subprojects,
+         this is used to call this function recursively.
+
+        :returns: A tuple with: project, version, and file name.
+        """
+        # Multiversion project.
+        if not parent_project.single_version:
+            response = self._match_multiversion_project(
+                parent_project=parent_project,
+                path=path,
+            )
+            if response:
+                return response
+
+        # Subprojects.
+        if check_subprojects:
+            response = self._match_subproject(
+                parent_project=parent_project,
+                path=path,
+            )
+            if response:
+                return response
+
+        # Single version project.
+        if parent_project.single_version:
+            response = self._match_single_version_project(
+                parent_project=parent_project,
+                path=path,
+            )
+            if response:
+                return response
+
+        return None, None, None
+
+    @staticmethod
+    def get_domain_from_host(host):
+        """
+        Get the normalized domain from a hostname.
+
+        A hostname can include the port.
+        """
+        return host.lower().split(":")[0]
+
+    # TODO: make this a private method once
+    # proxito uses the unresolve method directly.
+    def unresolve_domain(self, domain):
+        """
+        Unresolve domain by extracting relevant information from it.
+
+        :param str domain: Domain to extract the information from.
+        :returns: A tuple with: the project slug, domain object, and if the domain
+         is from an external version.
+        """
+        public_domain = self.get_domain_from_host(settings.PUBLIC_DOMAIN)
+        external_domain = self.get_domain_from_host(
+            settings.RTD_EXTERNAL_VERSION_DOMAIN
         )
-        return UnresolvedObject(final_project, lang_slug, version_slug, filename, parsed.fragment)
 
+        subdomain, *root_domain = domain.split(".", maxsplit=1)
+        root_domain = root_domain[0] if root_domain else ""
+
+        if public_domain in domain:
+            # Serve from the PUBLIC_DOMAIN, ensuring it looks like `foo.PUBLIC_DOMAIN`.
+            if public_domain == root_domain:
+                project_slug = subdomain
+                log.info("Public domain.", domain=domain)
+                return project_slug, None, False
 
-class Unresolver(SettingsOverrideObject):
+            # TODO: This can catch some possibly valid domains (docs.readthedocs.io.com)
+            # for example, but these might be phishing, so let's ignore them for now.
+            log.warning("Weird variation of our domain.", domain=domain)
```

Comment:
I took a look at this at New Relic and it seems we don't have any log for this: https://onenr.io/0oQD4oYZ5Qy 👍🏼